doc/cephfs/mount.rst

```diff
@@ -0,0 +1,69 @@
+Mounting CephFS
+===============
+
+You can use CephFS by mounting it to your local filesystem or by using
+cephfs-shell. CephFS can be mounted using kernel as well as using FUSE. Both
+have their own advantages. Read the following section to understand more
+about both of these ways to mount CephFS.
+
+Which CephFS Client?
+--------------------
+
+The FUSE client is the most accessible and the easiest to upgrade to the
+version of Ceph used by the storage cluster, while the kernel client will
+often give better performance.
+
+The clients do not always provide equivalent functionality, for example
+the fuse client supports client-enforced quotas while the kernel client
+does not.
+
+When encountering bugs or performance issues, it is often instructive to
+try using the other client, in order to find out whether the bug was
+client-specific or not (and then to let the developers know).
+
+General Pre-requisite for Mounting CephFS
+-----------------------------------------
+Before mounting CephFS, ensure that the client host (where CephFS has to be
+mounted and used) has a copy of the Ceph configuration file (i.e.
+``ceph.conf``) and a keyring of the CephX user that has CAPS for a Ceph MDS.
+Both of these files must already be present on the host where the Ceph MON
+resides.
+
+#. Generate a minimal conf file for the client host and place it at a
+   standard location::
+
+    # on client host
+    mkdir /etc/ceph
+    ssh {user}@{mon-host} "sudo ceph config generate-minimal-conf" | sudo tee /etc/ceph/ceph.conf
+
+   Alternatively, you may copy the conf file. But the above method generates
+   a conf with minimal details which is usually sufficient. For more
+   information, see :ref:`bootstrap-options`.
+
+#. Ensure that the conf has appropriate permissions::
+
+    chmod 644 /etc/ceph/ceph.conf
+
+#. Create a CephX user and get its secret key::
+
+    ssh {user}@{mon-host} "sudo ceph fs authorize cephfs client.foo / rw" | sudo tee /etc/ceph/ceph.client.foo.keyring
+
+   In above command, replace ``cephfs`` with the name of your CephFS, ``foo``
+   by the name you want for your CephX user and ``/`` by the path within your
+   CephFS for which you want to allow access to the client host and ``rw``
+   stands for both read and write permissions. Alternatively, you may copy the
```

Comment:
Now would be a good time to link to `client-auth`.

Reply:
Linked.